ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -55,22 +64,134 @@ object EndpointsCompanion {
       case NonFatal(t) => ServerError(t.description)
     }
   }
-
+  private type ET[A] = EitherT[Future, Unauthorized, A]
   trait ParsePayload[A] {
-    def parsePayload(jwt: DecodedJwt[String]): Unauthorized \/ A
+    def parsePayload(
+        jwt: DecodedJwt[String]
+    ): Error \/ A
+  }
+
+  trait CreateFromUserToken[A] {
+    def apply(
+        jwt: DecodedJwt[String],
+        listUserRights: UserId => Future[Vector[UserRight]],
+        getLedgerId: () => Future[String],
+    ): ET[A]
+  }
+
+  object CreateFromUserToken {
+
+    import com.daml.http.util.FutureUtil.either
+
+    trait FromUser[A, B] {
+      def apply(userId: String, actAs: List[String], readAs: List[String], ledgerId: String): A \/ B
+    }
+
+    private def transformUserTokenTo[B](
+        jwt: DecodedJwt[String],
+        listUserRights: UserId => Future[Vector[UserRight]],
+        getLedgerId: () => Future[String],
+    )(
+        fromUser: FromUser[Unauthorized, B]
```

Comment:
rather than defining a trait for this call only to then call it exactly once at the end maybe just kill the trait and call the function outside of `transformUserTokenTo`?